Indicators/PivotPointsHighLow.cs

```diff
@@ -0,0 +1,247 @@
+ * limitations under the License.
+*/
+
+using System;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// Pivot Points (High/Low), also known as Bar Count Reversals, indicator.
+    /// https://www.fidelity.com/learning-center/trading-investing/technical-analysis/technical-indicator-guide/pivot-points-high-low
+    /// </summary>
+    public class PivotPointsHighLow : WindowIndicator<IBaseDataBar>
+    {
+        private readonly int _lengthHigh;
+        private readonly int _lengthLow;
+
+        // Indicator will keep information of the last 100 pivot points
+        private readonly RollingWindow<PivotPoint> _pivotPoints = new RollingWindow<PivotPoint>(100);
+
+        /// <summary>
+        /// Event informs of new pivot point formed with new data update
+        /// </summary>
+        public event EventHandler<PivotPointsEventArgs> NewPivotPointFormed;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="PivotPointsHighLow"/> indicator with an equal high and low length
+        /// </summary>
+        /// <param name="length">The length parameter here defines the number of surround bars that we compare against the current bar high and lows for the max/min </param>
+        public PivotPointsHighLow(int length)
+            : this($"PivotPointsHighLow({length})", length, length)
+        { }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="PivotPointsHighLow"/> indicator
+        /// </summary>
+        /// <param name="lengthHigh">The number of surrounding bars whose high values should be less than the current one's for the high pivot point be assigned</param>
+        /// <param name="lengthLow">The number of surrounding bars whose lows values should be less than the current one's for the low pivot point be assigned</param>
+        public PivotPointsHighLow(int lengthHigh, int lengthLow)
+            : this($"PivotPointsHighLow({lengthHigh},{lengthLow})", lengthHigh, lengthLow)
+        { }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="PivotPointsHighLow"/> indicator
+        /// </summary>
+        /// <param name="name">The name of an indicator</param>
+        /// <param name="lengthHigh">The number of surrounding bars whose high values should be less than the current one's for the high pivot point be assigned</param>
+        /// <param name="lengthLow">The number of surrounding bars whose lows values should be less than the current one's for the low pivot point be assigned</param>
+        public PivotPointsHighLow(string name, int lengthHigh, int lengthLow) :
+            base(name, 2 * Math.Max(lengthLow, lengthHigh) + 1)
+        {
+            _lengthHigh = lengthHigh;
+            _lengthLow = lengthLow;
+        }
+
+        /// <inheritdoc />
+        protected override decimal ComputeNextValue(IReadOnlyWindow<IBaseDataBar> window, IBaseDataBar input)
```

Comment:
nit - Sorry  `/// <inheritdoc />` isn't supported at the moment by our doc builder, let's replace it for the full `<summary>`